src/fnft_nsev.c

```diff
@@ -69,72 +63,99 @@ UINT fnft_nsev_max_K(const UINT D, fnft_nsev_opts_t const * const opts)
-    const UINT deg,
-    const INT W,
-    COMPLEX * const transfer_matrix,
-    REAL const * const T,
-    const UINT D,
-    REAL const * const XI,
-    const UINT M,
-    COMPLEX *result,
-    fnft_nsev_opts_t * const opts);
-
-static inline INT tf2boundstates(
-    UINT D,
-    COMPLEX const * const q,
-    const UINT deg,
-    COMPLEX * const transfer_matrix,
-    REAL const * const T,
-    const REAL eps_t,
-    UINT * const K_ptr,
-    COMPLEX * const bound_states,
-    fnft_nsev_opts_t * const opts);
-
-static inline INT tf2normconsts_or_residues(
-    const UINT D,
-    COMPLEX const * const q,
-    REAL const * const T,
-    const UINT K,
-    COMPLEX * const transfer_matrix,
-    const UINT deg,
-    COMPLEX * const bound_states,
-    COMPLEX * const normconsts_or_residues,
-    fnft_nsev_opts_t * const opts);
-
-static inline INT refine_roots_newton(
-    const UINT D,
-    COMPLEX const * const q,
-    REAL const * const T,
-    const UINT K,
-    COMPLEX * bound_states,
-    nse_discretization_t discretization,
-    const UINT niter);
+
+static inline INT nsev_compute_boundstates(
+        UINT D,
+        COMPLEX const * const q,
+        COMPLEX * r,
+        const UINT deg,
+        COMPLEX * const transfer_matrix,
+        REAL const * const T,
+        const REAL eps_t,
+        UINT * const K_ptr,
+        COMPLEX * const bound_states,
+        fnft_nsev_opts_t * const opts);
+
+static inline INT fnft_nsev_base(
+        const UINT D,
+        COMPLEX * const q,
+        COMPLEX * r,
+        REAL const * const T,
+        const UINT M,
+        COMPLEX * const contspec,
+        REAL const * const XI,
+        UINT * const K_ptr,
+        COMPLEX * const bound_states,
+        COMPLEX * const normconsts_or_residues,
+        const INT kappa,
+        fnft_nsev_opts_t *opts);
+
+static inline INT nsev_compute_contspec(
+        const UINT deg,
+        const INT W,
+        COMPLEX * const transfer_matrix,
+        COMPLEX const * const q,
+        COMPLEX * r,
+        REAL const * const T,
+        const UINT D,
+        REAL const * const XI,
+        const UINT M,
+        COMPLEX * const result,
+        const INT kappa,
+        fnft_nsev_opts_t * const opts);
+
+static inline INT nsev_compute_normconsts_or_residues(
+        const UINT D,
+        COMPLEX const * const q,
+        COMPLEX * r,
+        REAL const * const T,
+        const UINT K,
+        COMPLEX * const bound_states,
+        COMPLEX * const normconsts_or_residues,
+        fnft_nsev_opts_t * const opts);
+
+static inline INT nsev_refine_bound_states_newton(const UINT D,
+        COMPLEX const * const q,
+        COMPLEX * r,
+        REAL const * const T,
+        UINT K,
+        COMPLEX * bound_states,
+        nse_discretization_t discretization,
+        FNFT_UINT niter);
 
 /**
  * Fast nonlinear Fourier transform for the nonlinear Schroedinger
  * equation with vanishing boundary conditions.
  * See the header file for documentation.
```

Comment:
I think it could be helpful to add a short comment about this specific function fits into the general scheme (takes care of Richardson and any required down/resampling before calling fnft_nsev_base).